espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/Finder.java

```diff
@@ -73,49 +50,41 @@ public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> u
             // If we have a match, return success
             Element element = new Element(matcher);
             response.setValue(element);
-            response.setSessionId(uriParams.get("sessionId"));
             return response;
         } catch (InvalidStrategyException e) {
             return new BadRequestResponse(e.getMessage());
-        } catch (ServerErrorException e) {
-            return new InternalErrorResponse(e.getMessage());
         } catch (ElementNotFoundException e) {
             return new BadRequestResponse("Could not find element with " + strategy.getStrategyName() + ": " + selector);
         }
     }
 
     ///Find By different strategies
-    private ViewInteraction findBy(Strategy strategy, String selector) throws ServerErrorException, InvalidStrategyException, ElementNotFoundException {
+    private ViewInteraction findBy(Strategy strategy, String selector) throws InvalidStrategyException, ElementNotFoundException {
         ViewInteraction matcher;
 
-        try {
-            switch (strategy) {
-                case ID: // with ID
-
-                    // find id from target context
-                    int id = InstrumentationRegistry.getTargetContext().getResources().getIdentifier(selector, "Id",
-                            InstrumentationRegistry.getTargetContext().getPackageName());
-                    matcher = onView(withId(id));
-                    break;
-                case CLASS_NAME:
-                    // with class name
-                    // TODO: improve this finder with instanceOf
-                    matcher = onView(withClassName(endsWith(selector)));
-                    break;
-                case TEXT:
-                    // with text
-                    matcher = onView(withText(selector));
-                    break;
-                case ACCESSIBILITY_ID:
-                    // with content description
-                    matcher = onView(withContentDescription(selector));
-                    break;
-                default:
-                    throw new InvalidStrategyException("Strategy is not implemented: " + strategy.getStrategyName());
-            }
-        }
-        catch (Exception e) {
-            throw new ServerErrorException();
+        switch (strategy) {
+            case ID: // with ID
+
+                // find id from target context
+                int id = InstrumentationRegistry.getTargetContext().getResources().getIdentifier(selector, "Id",
```

Comment:
I find it more readable if InstrumentationRegistry.getTargetContext() is assigned to some local variable first